src/lib/comms/sol-coap.c

```diff
@@ -1665,7 +1666,7 @@ network_event(void *data, const struct sol_network_link *link, enum sol_network_
 }
 
 static struct sol_coap_server *
-sol_coap_server_new_full(enum sol_socket_type type, const struct sol_network_link_addr *servaddr)
+sol_coap_server_new_full(enum sol_socket_default_type type, const struct sol_network_link_addr *servaddr)
```

Comment:
no, this need to be changed to receive a pre-configured `struct sol_socket`. Then you do the setup elsewhere, like ble/iotivity, udp, etc.